### ISSUE TITLE
Date parsing issues in Laravel 5.4

### ISSUE THREAD

**Issue by DFurnes** [DoSomething/northstar#689]
Looks like we're still having a few issues with date serialization in Laravel 5.4. From the Thor logs:

```
[2017-11-06 21:36:07] production.ERROR: Exception: DateTime::__construct(): Failed to parse time string (1510003920000) at position 11 (0): Unexpected character in /var/www/northstar/releases/20171106201243/vendor/nesbot/carbon/src/Carbon/Carbon.php:291
Stack trace:
#0 /var/www/northstar/releases/20171106201243/vendor/nesbot/carbon/src/Carbon/Carbon.php(291): DateTime->__construct('1510003920000', Object(DateTimeZone))
#1 /var/www/northstar/releases/20171106201243/app/helpers.php(48): Carbon\Carbon->__construct(Object(MongoDB\BSON\UTCDateTime))
#2 /var/www/northstar/releases/20171106201243/app/helpers.php(66): format_date(Object(MongoDB\BSON\UTCDateTime), 'Y-m-d\\TH:i:sP')
#3 /var/www/northstar/releases/20171106201243/app/Models/User.php(413): iso8601(Object(MongoDB\BSON\UTCDateTime))
#4 /var/www/northstar/releases/20171106201243/app/Providers/AppServiceProvider.php(51): Northstar\Models\User->toBlinkPayload()
#5 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php(348): Northstar\Providers\AppServiceProvider->Northstar\Providers\{closure}(Object(Northstar\Models\User))
#6 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php(199): Illuminate\Events\Dispatcher->Illuminate\Events\{closure}('eloquent.update...', Array)
#7 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php(172): Illuminate\Events\Dispatcher->dispatch('eloquent.update...', Array, false)
#8 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasEvents.php(148): Illuminate\Events\Dispatcher->fire('eloquent.update...', Object(Northstar\Models\User))
#9 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(597): Illuminate\Database\Eloquent\Model->fireModelEvent('updated', false)
#10 /var/www/northstar/releases/20171106201243/app/Models/Model.php(95): Illuminate\Database\Eloquent\Model->performUpdate(Object(Jenssegers\Mongodb\Eloquent\Builder), Array)
#11 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(515): Northstar\Models\Model->performUpdate(Object(Jenssegers\Mongodb\Eloquent\Builder))
#12 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Auth/EloquentUserProvider.php(86): Illuminate\Database\Eloquent\Model->save()
#13 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Auth/SessionGuard.php(534): Illuminate\Auth\EloquentUserProvider->updateRememberToken(Object(Northstar\Models\User), 'GTOaxS7v5TQuZx7...')
#14 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Auth/SessionGuard.php(494): Illuminate\Auth\SessionGuard->cycleRememberToken(Object(Northstar\Models\User))
#15 /var/www/northstar/releases/20171106201243/app/Http/Controllers/Web/AuthController.php(164): Illuminate\Auth\SessionGuard->logout()
#16 [internal function]: Northstar\Http\Controllers\Web\AuthController->getLogout(Object(Illuminate\Http\Request))
#17 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(55): call_user_func_array(Array, Array)
#18 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(44): Illuminate\Routing\Controller->callAction('getLogout', Array)
#19 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Route.php(203): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Northstar\Http\Controllers\Web\AuthController), 'getLogout')
#20 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Route.php(160): Illuminate\Routing\Route->runController()
#21 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Router.php(572): Illuminate\Routing\Route->run()
#22 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(30): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#23 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(41): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#24 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
#25 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#26 /var/www/northstar/releases/20171106201243/app/Http/Middleware/SetLanguageFromHeader.php(33): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#27 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Northstar\Http\Middleware\SetLanguageFromHeader->handle(Object(Illuminate\Http\Request), Object(Closure))
#28 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#29 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(65): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#30 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure))
#31 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#32 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#33 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#34 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#35 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#36 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#37 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#38 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#39 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#40 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#41 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(59): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#42 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#43 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#44 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#45 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Router.php(574): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#46 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Router.php(533): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#47 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Router.php(511): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#48 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(176): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#49 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(30): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#50 /var/www/northstar/releases/20171106201243/vendor/fideloper/proxy/src/TrustProxies.php(56): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#51 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Fideloper\Proxy\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
#52 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#53 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(30): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#54 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#55 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#56 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#57 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle(Object(Illuminate\Http\Request), Object(Closure))
#58 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#59 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(30): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#60 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#61 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#62 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(46): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#63 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(148): Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#64 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#65 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#66 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(151): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#67 /var/www/northstar/releases/20171106201243/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#68 /var/www/northstar/releases/20171106201243/public/index.php(51): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#69 {main}
```

**Comment by DFurnes** [comment]
So this is because we're passing a raw Mongo `UTCDateTime` to our formatting helpers, and Carbon is unable to automatically parse that into a DateTime instance since it has milliseconds in the timestamp value. Adding these fields to the `$dates` property on the model clears up the problem.

**Comment by DFurnes** [comment]
This should be fixed by #690.
